executor/builder.go

```diff
@@ -2177,6 +2185,22 @@ func markChildrenUsedCols(outputSchema *expression.Schema, childSchema ...*expre
 	return
 }
 
+// extractChildrenUsedColIdxs extract column indexes of child executors from children used columns mark
+func extractChildrenUsedColIdxs(childrenUsedColsMark [][]bool) [][]int {
```

Comment:
When will `childrenUsedColsMark ` be of type `[][]int`?

Reply:
This is to maintain the same dimension as childrenUsedColsMark

Reply:
We would not pass a `[][]bool` into this function.
I think `[]bool` is enough

Reply:
> We would not pass a `[][]bool` into this function.
> I think `[]bool` is enough

Yes, you are right. I think `extractChildrenUsedColIdxs` method return value keep the same dimension as `childrenUsedColsMark` may provide more scenarios for code reuse. But I also can change it to []int.

Reply:
The return value can also be defined as `[]int`

Reply:
Yes, in the next discussion about `newJoiner`, inside the [block](https://github.com/pingcap/tidb/blob/master/executor/joiner.go#L142-L147) will call `extractChildrenUsedColIdxs` twice.